enos/provider/g5k.py

```diff
@@ -165,35 +167,88 @@ def _make_reservation(self):
         criteria = {}
         # Actual criteria are :
         # - Number of node per site
+        queues = self._get_queues()
+        for queue, clusters in queues.iteritems():
+            criteria[queue] = {}
+
+            for cluster, roles in self.config["resources"].items():
+                if not cluster in clusters:
+                    continue
+
+                site = get_cluster_site(cluster)
+                nb_nodes = reduce(operator.add, map(int, roles.values()))
+                criterion = "{cluster='%s'}/nodes=%s" % (cluster, nb_nodes)
+                criteria[queue].setdefault(site, []).append(criterion)
+
+        # Add a vlan to the first queue that contains a matching site
+        for site, vlan in self.config["vlans"].items():
+            for queue, qc in criteria.iteritems():
+                if site in qc.keys():
+                    criteria[queue][site].append(vlan)
+                    break
+
+        # Make one reservation per queue
+        self.gridjobs = []
+        for queue in queues:
+            # Compute the specification for the reservation
+            name = "%s-%d" % (self.config['name'], len(self.gridjobs))
+            jobs_specs = [(OarSubmission(resources='+'.join(c),
+                                         name=name,
+                                         queue=queue), s)
+                          for s, c in criteria[queue].items()]
+            logging.info("Criteria for the reservation: %s" % pf(jobs_specs))
+
+            # Make the reservation
+            gridjob, _ = EX5.oargridsub(
+                jobs_specs,
+                reservation_date=self.config['reservation'],
+                walltime=self.config['walltime'].encode('ascii', 'ignore'),
+                job_type='deploy',
+                queue=queue)
+
+            # TODO - move this upper to not have a side effect here
+            if gridjob is not None:
+                self.gridjobs.append(gridjob)
+                logging.info("Using new oargrid job %s" % gridjob)
+            else:
+                logging.error("No oar job was created.")
+                if len(self.gridjobs) > 0:
+                    EX5.oargriddel(self.gridjobs)
+                    logging.error("Deleting already submitted jobs: %s" % self.gridjobs)
+                sys.exit(26)
+
+
+    def _get_queues(self):
```

Comment:
I agree but I think this should be a different PR, because either I change a the signature of a few methods (and then the class will be inconsistent) or I do it for all the methods (but then I touch many lines that are not related to this PR).